apps/src/lab2/views/dialogs/DialogControlContext.ts

```diff
@@ -1,14 +1,29 @@
 import React, {useContext} from 'react';
 
-import {TypedDialogProps, closeDialogType} from './types';
+import {
+  getDeferredPromise,
+  DeferredPromiseObject,
+} from '@cdo/apps/lab2/utils/getDeferredPromise';
+
+import {
+  TypedDialogProps,
+  DialogCloseFunctionType,
+  DialogClosePromiseReturnType,
+} from './types';
 
 interface DialogControl {
-  closeDialog: closeDialogType;
-  showDialog: (args: TypedDialogProps) => void;
+  closeDialog: DialogCloseFunctionType;
+  showDialog: (args: TypedDialogProps) => Promise<DialogClosePromiseReturnType>;
+  deferredPromiseObject: DeferredPromiseObject;
+  promiseArgs: unknown;
+  setPromiseArgs: (args: unknown) => void;
 }
 
 export const DialogControlContext = React.createContext<DialogControl>({
   closeDialog: () => {},
-  showDialog: () => {},
+  showDialog: () => Promise.resolve({type: 'cancel'}),
+  deferredPromiseObject: getDeferredPromise(),
+  promiseArgs: undefined,
+  setPromiseArgs: () => {},
```

Comment:
`promiseArgs` will be handed into the `deferredPromiseObject` when it's resolved/rejected at the end.

So from your dialog, you can do this:
```

const SomeCustomDialog = () => {
  const {promiseArgs, setPromiseArgs} = useDialogControl();
  
  // promiseArgs is unknown. Let's assume you're using it as a string.
  const prompt = promiseArgs as string;
  // display a text box. When the user types, update the promise args. When the dialog closes, the value of
  // the typed in value will be handed through.
  return <input type = 'text' value = {prompt} onChange = {e => setPromiseArgs(e.target.value) } />
}

The final values handed through to the promise when it's resolved are:

```
{
  type : 'confirm' | 'neutral' | 'cancel' // the button that was pressed
  args : ... // unknown type. Put in whatever you'd like!
}
```